src/conn/conn_layered.c

```diff
@@ -1276,3 +1276,173 @@ __wt_disagg_advance_checkpoint(WT_SESSION_IMPL *session, bool ckpt_success)
 
     return (0);
 }
+
+/*
+ * __layered_move_updates --
+ *     Move the updates of a key to the stable table
+ */
+static int
+__layered_move_updates(WT_CURSOR_BTREE *cbt, WT_ITEM *key, WT_UPDATE *upds)
+{
+    /* Search the page. */
+    WT_RET(__wt_row_search(cbt, key, true, NULL, false, NULL));
+
+    /* Apply the modification. */
+    WT_RET(__wt_row_modify(cbt, key, NULL, &upds, WT_UPDATE_INVALID, false, false));
+
+    return (0);
+}
+
+/*
+ * __layered_drain_ingest_table --
+ *     Moving all the data from a single ingest table to the corresponding stable table
+ */
+static int
+__layered_drain_ingest_table(WT_SESSION_IMPL *session, WT_LAYERED_TABLE_MANAGER_ENTRY *entry)
+{
+    WT_CURSOR *stable_cursor, *version_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_DECL_ITEM(key);
+    WT_DECL_ITEM(tmp_key);
+    WT_DECL_ITEM(value);
+    WT_DECL_RET;
+    WT_TIME_WINDOW tw;
+    WT_UPDATE *prev_upd, *tombstone, *upd, *upds;
+    uint8_t flags, location, prepare, type;
+    int cmp;
+    char buf[256];
+    const char *cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL, NULL, NULL};
+
+    WT_RET(__layered_table_get_constituent_cursor(session, entry->ingest_id, &stable_cursor));
+    cbt = (WT_CURSOR_BTREE *)stable_cursor;
+    WT_RET(__wt_snprintf(buf, sizeof(buf),
+      "debug=(dump_version=(enabled=true,visible_only=true,start_timestamp=%" PRIx64 "))",
+      S2C(session)->txn_global.stable_timestamp));
+    cfg[1] = buf;
+    WT_RET(__wt_open_cursor(session, entry->ingest_uri, NULL, cfg, &version_cursor));
+    /* We only care about binary data. */
+    F_SET(version_cursor, WT_CURSTD_RAW);
+
+    prev_upd = tombstone = upd = upds = NULL;
+
+    WT_ERR(__wt_scr_alloc(session, 0, &key));
+    WT_ERR(__wt_scr_alloc(session, 0, &tmp_key));
+    WT_ERR(__wt_scr_alloc(session, 0, &value));
+
+    for (;;) {
+        tombstone = upd = NULL;
+        WT_ERR_NOTFOUND_OK(version_cursor->next(version_cursor), true);
+        if (ret == WT_NOTFOUND) {
+            if (key->size > 0 && upds != NULL)
+                WT_ERR(__layered_move_updates(cbt, key, upds));
```

Comment:
Should this also set `upds = NULL`, just like below?

Reply:
yes, good catch